content/blog/rag-pipelines-from-scratch/index.md

```diff
@@ -0,0 +1,423 @@
+
+```python
+from haystack.preview.components.builders.prompt_builder import PromptBuilder
+
+prompt_builder = PromptBuilder(template="What's the official language of {{ country }}?")
+prompt_builder.run(country="France")
+# returns {'prompt': "What's the official language of France?"}
+```
+
+Note how we defined a variable, `country`, by wrapping its name in double curly brackets. PromptBuilder lets you define any input variable that way: if the prompt template was `"What's the official language of {{ nation }}?"`, the `run()` method of `PromptBuilder` would have expected a `nation` input.
+
+This syntax comes from [Jinja2](https://jinja.palletsprojects.com/en/3.0.x/intro/), a popular templating library for Python. If you have ever used Flask, Django, or Ansible, you will feel at home with `PromptBuilder`. Instead, if you never heard of any of these libraries, you can check out the [syntax](https://jinja.palletsprojects.com/en/3.0.x/templates/) on Jinja's documentation. Jinja has a powerful templating language and offers way more features than you'll ever need in prompt templates, ranging from simple if statements and for loops to object access through dot notation, nesting of templates, variables manipulation, macros, full-fledged import and encapsulation of templates, and more.
+
+# A Simple Generative Pipeline
+
+With these two components, we can assemble a minimal pipeline to see how they work together. Connecting them is trivial: `PromptBuilder` generates a `prompt` output, and `GPTGenerator` expects an input with the same name and type.
+
+```python
+from haystack.preview import Pipeline
+from haystack.preview.components.generators.openai.gpt import GPTGenerator
+from haystack.preview.components.builders.prompt_builder import PromptBuilder
+
+pipe = Pipeline()
+pipe.add_component("prompt_builder", PromptBuilder(template="What's the official language of {{ country }}?"))
+pipe.add_component("llm", GPTGenerator(api_key=api_key))
+pipe.connect("prompt_builder", "llm")
+
+pipe.run({"prompt_builder": {"country": "France"}})
+# returns {"llm": {"replies": ['The official language of France is French.'] }}
+```
+
+Here is the pipeline graph:
+
+![Simple LLM pipeline](simple-llm-pipeline.png)
+
+# Make the LLM cheat
+
+Building the Generative part of a RAG application was very simple! So far, we only provided the question to the LLM, but no information to base its answers on. Nowadays, LLMs possess a lot of general knowledge, so questions about famous countries such as France or Germany are easy for them to reply to correctly. However, when using an app about world countries, some users may be interested in knowing more about obscure or defunct microstates that don't exist anymore. In this case, ChatGPT is unlikely to provide the correct answer without any help.
```

Comment:
Even if we're talking about the model at a quite high level? Changing it to `gpt-3.5-turbo` may look a bit jarring here. Or did you mean to replace it with "GPT 3.5" instead of "ChatGPT"?